python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -566,6 +569,27 @@ def test_agg_with_two_agg_exprs(self):
             self.spark.read.table(self.tbl_name).agg({"name": "min", "id": "max"}).toPandas(),
         )
 
+    def test_write_operations(self):
+        with tempfile.TemporaryDirectory() as d:
+            df = self.connect.range(1, 100)
+            df.write.mode("overwrite").format("csv").save(d)
+
+            ndf = self.connect.read.load(d, format="csv")
+            df.toPandas().equals(ndf.toPandas())
```

Comment:
we'd better use `self.assertEqual` instead, otherwise it will be painful to debug if it fails

Reply:
Done

Reply:
it's actually `self.assert_eq` which is a custom function provided in the test suite

Reply:
I'm using collect now to avoid some data type issues.